keras/utils/np_utils.py

```diff
@@ -15,11 +17,16 @@ def to_categorical(y, num_classes=None):
         y: class vector to be converted into a matrix
             (integers from 0 to num_classes).
         num_classes: total number of classes.
+        dtype: The data type expected by the input, as a string
+            (`float32`, `float64`, `int32`...)
 
     # Returns
         A binary matrix representation of the input. The classes axis
         is placed last.
     """
+    if not dtype:
+        dtype = K.floatx()
```

Comment:
We should not make `np_utils` depend on `backend`, so please hardcode the default `'float32'` in the signature. It's not going to break if there's a mismatch with `floatx` anyway since we cast Numpy data when feeding it to models.

Reply:
@fchollet OK, I agree on your point on `backend` dependency. But why hardcode? Even if we hardcode, I think `int8` is a better option as it occupies 3/4 less memory -- we currently constantly suffer from memory issues due to this function.

Reply:
@fchollet Sorry I misunderstood your comments. I'll fix this.